OrbitCore/SymbolHelper.cpp

```diff
@@ -0,0 +1,166 @@
+#include "SymbolHelper.h"
+
+#include <fstream>
+
+#include "ElfFile.h"
+#include "OrbitBase/Logging.h"
+#include "Path.h"
+#include "Pdb.h"
+
+std::vector<std::string> SymbolHelper::collector_symbol_locations = {
+    "/home/cloudcast/",  "/home/cloudcast/debug_symbols/",
+    "/mnt/developer/",   "/mnt/developer/debug_symbols/",
+    "/srv/game/assets/", "/srv/game/assets/debug_symbols/"};
+
+std::vector<std::string> SymbolHelper::visualizer_symbol_locations;
+
+bool LoadFromElfFile(std::shared_ptr<Module> module,
+                     const std::unique_ptr<ElfFile>& elf_file) {
+  auto load_bias = elf_file->GetLoadBias();
+  if (!load_bias) {
+    ERROR(
+        "Unable to get load_bias %s (does the file have PT_LOAD program "
+        "headers?)",
+        elf_file->GetFilePath().c_str());
+    return false;
+  }
+
+  std::shared_ptr<Pdb> pdb =
+      std::make_shared<Pdb>(module->m_AddressStart, load_bias.value(),
+                            elf_file->GetFilePath(), module->m_FullName);
+
+  if (!elf_file->GetFunctions(pdb.get())) {
+    ERROR("Unable to load functions from %s", elf_file->GetFilePath().c_str());
+    return false;
+  }
+
+  pdb->ProcessData();
+
+  module->m_Pdb = pdb;
+  module->m_PdbName = elf_file->GetFilePath();
+  module->SetLoaded(true);
+  return true;
+}
+
+bool FindAndLoadSymbols(std::shared_ptr<Module> module,
+                        const std::vector<std::string> search_directories) {
+  FAIL_IF(module->m_DebugSignature.empty(), "build id is empty of module %s",
+          module->m_Name.c_str());
+  std::string name_without_extension = Path::StripExtension(module->m_Name);
+
+  std::vector<std::string> search_file_paths;
+  for (const auto& directory : search_directories) {
+    search_file_paths.emplace_back(directory + name_without_extension +
+                                   ".debug");
+    search_file_paths.emplace_back(directory + module->m_Name + ".debug");
```

Comment:
could the module be named the same but just placed to symbols/ (sub)folder, in other words - shouldn't this check the m_Name too?

Reply:
To be even more correct, the binary should contain a gnu_debuglink section which specifies the exact name of the symbols file. This is tracked in a different feature with lower priority. 

I would leave it for now, which means its currently a requirement to have the file named `<module>.debug` (This mirrors whats in Development.md)

Reply:
.. or you could add m_name here and update Development.md

Reply:
done